src/premint/ZoraCreator1155Attribution.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IMinter1155} from "../interfaces/IMinter1155.sol";
+import {IZoraCreator1155} from "../interfaces/IZoraCreator1155.sol";
+import {ICreatorRoyaltiesControl} from "../interfaces/ICreatorRoyaltiesControl.sol";
+import {ECDSAUpgradeable} from "@zoralabs/openzeppelin-contracts-upgradeable/contracts/utils/cryptography/ECDSAUpgradeable.sol";
+import {ZoraCreatorFixedPriceSaleStrategy} from "../minters/fixed-price/ZoraCreatorFixedPriceSaleStrategy.sol";
+
+struct ContractCreationConfig {
```

Comment:
this doesn't quite belong in this file, as it is primarily used to create new contracts and get a deterministic address based on these fields